### DISCUSSION TITLE
小程序正式版的发布标准

### DISCUSSION THREAD

**Discussion by LeeJim** [Tencent/tdesign-miniprogram#465] [Show and tell]
欢迎大家讨论。

首先，说下我认为的标准有：
- API 实现与文档一致
- 基础组件都有
- 完备的单元测试

**Comment by LeeJim** [comment]
示例代码改造：保证示例与代码是一致的

**Comment by jkiss** [comment]
可自定义theme

**Reply by LeeJim** [reply]
必须的

**Comment by LeeJim** [comment]
1.0 正式版本预备做以下改动过：

- [ ] 组件文件名从 `组件名` -> `index`
- [x] 插槽命名统一使用 `kebab-case` 而不是 `camel-case` 
- [x] 全部支持 CSS Variables，可自定义主题 
- [x] 每个组件单独目录，主要是改动父子组件同目录的问题
- [x] 开启 `virtualHost = true`，不再使用 `shadow-dom`；可能会导致基础版本升级至 2.20+

**Reply by LeeJim** [reply]
#1273 完成【全部支持 CSS Variables，可自定义主题】
#1277 完成【插槽命名统一使用 kebab-case 而不是 camel-case】

**Comment by xiaodongning** [comment]
正式版预计什么时候发布

**Reply by LeeJim** [reply]
2022.2.28 已发布正式版本
